File: JavaAlgorithm/src/main/java/algo/tzashinorpu/ThirdRound/Chapter07/invertTree_226.java
package algo.tzashinorpu.ThirdRound.Chapter07;

import algo.tzashinorpu.ThirdRound.TreeNode;

public class invertTree_226 {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode left = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(left);
        return root;
    }
}

File: Clion/ZeroTrac/Tree/invertTree_226.cpp
//
// Created by TzashiNorpu on 10/31/2022.
//
#include "tree.h"

using namespace Tree;

TreeNode *Solution::invertTree(TreeNode *root) {
  if (!root) return nullptr;
  TreeNode *l = invertTree(root->left);
  TreeNode *r = invertTree(root->right);
  root->left = r;
  root->right = l;
  return root;
}